eva-accession-pipeline/src/main/java/uk/ac/ebi/eva/accession/pipeline/steps/processors/ContigReplacerProcessor.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.accession.pipeline.steps.processors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemProcessor;
+import uk.ac.ebi.eva.accession.core.contig.ContigMapping;
+import uk.ac.ebi.eva.accession.core.contig.ContigSynonyms;
+import uk.ac.ebi.eva.commons.core.models.IVariant;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+import uk.ac.ebi.eva.commons.core.models.pipeline.VariantSourceEntry;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Converts the contig to it's GenBank synonym when possible. If the synonym can't be determined it keeps the contig as
+ * is
+ */
+public class ContigReplacerProcessor implements ItemProcessor<IVariant, IVariant> {
+
+    private static final Logger logger = LoggerFactory.getLogger(ContigReplacerProcessor.class);
+
+    private ContigMapping contigMapping;
+
+    private Set<String> processedContigs;
+
+    public ContigReplacerProcessor(ContigMapping contigMapping) {
+        this.contigMapping = contigMapping;
+        this.processedContigs = new HashSet<>();
+    }
+
+    @Override
+    public IVariant process(IVariant variant) throws Exception {
+        String contigName = variant.getChromosome();
+        ContigSynonyms contigSynonyms = contigMapping.getContigSynonyms(contigName);
+        boolean contigPresentInAssemblyReport = contigSynonyms != null;
+
+        if (!contigPresentInAssemblyReport) {
+            if (!processedContigs.contains(contigName)) {
+                logger.warn("Contig '" + variant.getChromosome() + "' was not found in the assembly report!");
+                processedContigs.add(contigName);
```

Comment:
what do others think about adding a method to log the total count of missing contigs? (also, split processedContigs in missingContigs and nonEquivalentContigs and log both):

```

    @AfterStep
    public ExitStatus afterStep(StepExecution stepExecution) {
        // declared as fields: Set<String> missingContigs, nonEquivalentContigs;
        if (!missingContigs.isEmpty()) {
            logger.warn("{} contigs were not present in the assembly report", missingContigs.size());
        }
        if (!nonEquivalentContigs.isEmpty()) {
            logger.warn("{} contigs were not replaced by INSDC names because the sequence was not identical",
                        nonEquivalentContigs.size());
        }
        return stepExecution.getExitStatus();
    }
```
(I just learnt that `@AfterStep` exists)

Reply:
We would need to register the `ContigReplacerProcessor` as a listener in the Step for the annotation to work because we are using a composite processor. Discussed offline with @jmmut and agreed not to do it.

https://stackoverflow.com/questions/39697529/spring-batch-use-before-and-after-step-methods-within-a-compositeitemwriter